src/gt4py/next/otf/compilation/build_systems/cmake_lists.py

```diff
@@ -30,22 +30,31 @@ class LinkDependency(eve.Node):
     target: str
 
 
+class Language(eve.Node):
+    name: str
+
+
 class CMakeListsFile(eve.Node):
     project_name: str
     find_deps: Sequence[FindDependency]
     link_deps: Sequence[LinkDependency]
     source_names: Sequence[str]
     bin_output_suffix: str
+    languages: Sequence[Language]
 
 
 class CMakeListsGenerator(eve.codegen.TemplatedGenerator):
     CMakeListsFile = as_jinja(
         """
-        project({{project_name}})
         cmake_minimum_required(VERSION 3.20.0)
+        
+        project({{project_name}})
 
         # Languages
-        enable_language(CXX)
+        if(NOT DEFINED CMAKE_CUDA_ARCHITECTURES)
```

Comment:
we could default to the arch of device 0 as we do in cartesian, see https://github.com/petiaccja/gt4py/blob/ecdba0124a0b834e56c22fb8b75a96c2488b8325/src/gt4py/cartesian/backend/pyext_builder.py#L40

Reply:
Did you check if `CMAKE_CUDA_ARCHITECTURES` does the right thing at compilation? I never used it (we never upgraded gridtools' cmake to work with it, because we just went through a big cmake refactoring when that feature was introduced).

Reply:
It sets the flags for `nvcc`: `--generate-code=arch=compute_60,code=[compute_60,sm_60]`. That looks correct to me.

Reply:
I'd scan the node for CUDA-compatible devices and either pick the oldest architecture or simply generate the binary for every specific architecture. The problem is, we don't really have infrastructure for specifying command line options to `cmake`, and it would be a mistake to implement this so low in the call stack.